triplea-game/triplea
# Inline DefaultUnitFactory

## Overview

Follow-up to [this discussion](https://github.com/triplea-game/triplea/pull/3764#discussion_r209090492).

The only usage of `DefaultUnitFactory` is in `TestGameLoader`, so this PR inlines the `DefaultUnitFactory` implementation there.

## Functional Changes

None.

## Refactoring Changes

* Removed `Serializable` from `IUnitFactory`.
* Renamed `IUnitFactory` to `UnitFactory`.

## Manual Testing Performed

Loaded a few older (within six months) save games to ensure there wasn't a `DefaultUnitFactory` instance lurking in there.

## Additional Review Notes

The fact that `IUnitFactory` is `Serializable` raises some questions as to whether `DefaultUnitFactory` can be removed, as well as if `IUnitFactory` can be refactored as done in this PR.  Hopefully, the following information will show this is the case:

* `DefaultUnitFactory` hasn't been used in production code since grid games were removed in 88040d3f2 (2015).  So, even if there was a save game out there with a serialized instance, it wouldn't be supported in the current release anyway.
* Even though `IUnitFactory` is serializable, the current code, at least, never stores a reference to a unit factory instance.  Thus, it doesn't appear possible for such an instance to be captured in an serialized object graph.  I think it was simply marked `Serializable` because `IGameLoader` is serializable, and the original author, possibly, thought game loader instances would store a unit factory instance in a field rather than creating a new instance upon each call to `getUnitFactory()`.
